### ISSUE TITLE
Is there a way to generate a .proto template with a Wire class?

### ISSUE THREAD

**Issue by changdaniel** [square/wire#2320]
Supporting some sort of API like:
`
System.out.println(Dinosaur.toProto());
`

I need to use the proto template for compatibility with existing projects or for debugging purposes.

**Comment by oldergod** [comment]
I don't see `printToProto` in the javadoc https://developers.google.com/protocol-buffers/docs/reference/java/com/google/protobuf/util/JsonFormat

Does it still exist?

**Comment by btiernay** [comment]
I had a similar question about being able to generate a `.proto` file from a `com.squareup.wire.schema.Schema` object. Is that possible today?

**Comment by JakeWharton** [comment]
Yes, the toString of Message and such try hard to produce a valid proto.

**Comment by btiernay** [comment]
> Yes, the toString of Message and such try hard to produce a valid proto.

Is that in response to me? My use case is parsing protos, making some transformation and then exporting them out. Thus, I'm using the `com.squareup.wire.schema.Schema` class instances in association with a `com.squareup.wire.schema.SchemaLoader`.

**Comment by JakeWharton** [comment]
Yes. Someone seems to have deleted my tool, but you can see #660 for an example of parsing a Schema, transforming it, and then producing a new set of proto files.

**Comment by oldergod** [comment]
The tool got removed when it got converted into a [ProtoTarget](https://github.com/square/wire/blob/e28efdf8e024b3947be4d488f19a1633b37ae755/wire-library/wire-compiler/src/main/java/com/squareup/wire/schema/Target.kt#L444-L497) to be used with the Gradle plugin.
